File: Selenium/Activities/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.color import Color

with webdriver.Firefox() as driver:

    driver.get("https://training-support.net/webelements/target-practice")
    print("Title of the page: ", driver.title)
    
    ##3rd header on the page and print it's text to the console.
    header3_message = driver.find_element(By.XPATH, "//h3[contains(text(), '#3')]").text
    print("message on header3: ", header3_message)

    ##5th header on the page and print it's color.
    element = Color.from_string(driver.find_element(By.XPATH, "//h5[contains(text(), '#5')]").value_of_css_property("color"))
    ##rgb_color = Color.from_string(element).value_of_CSS_property('color')
    print("RGB color: ", element)
    print("heading5 colour as Hexcode: ", element.hex)
    ##hex_color = Color.from_string(element).hex
    ##print("hex color is: ", hex_color)

    ##purple button and print all it's classes.
    Att_value = driver.find_element(By.XPATH, "//button[contains(text(), 'Pu')]").get_attribute("Class")
    print("Purple button Class attribute: ", Att_value)

    ##slate button and print it's text.
    button_text = driver.find_element(By.XPATH, "//button[contains(text(), 'Sl')]").text
    print("text on Slate button: ", button_text)

    driver.quit()

File: Selenium/Activities/Activity4.java
package Activity;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.Color;

public class Activity4 {
	
	public static void main(String[] args) {
		
		WebDriver driver = new FirefoxDriver();
		
		//Get the title of the page and print it to the console.
		driver.get("https://training-support.net/webelements/target-practice");
		System.out.println("Title of the page: " + driver.getTitle());
		
		//3rd header on the page and print it's text to the console
		
		String header3_message = driver.findElement(By.xpath("//h3[contains(text(), '#3')]")).getText();
		System.out.println("Header 3 message is: "  + header3_message);
		
		//5th header on the page and print it's color
		Color header5_color = Color.fromString(driver.findElement(By.xpath("//h5[contains(text(), '#5')]")).getCssValue("color"));
		System.out.println("Header5 color RGB: " + header5_color.asRgba());
		System.out.println("Header5 color as hexcode: " + header5_color.asHex());
		
		//purple button and print all it's classes
		WebElement element = driver.findElement(By.xpath("//button[contains(text(), 'Pu')]"));
		String att_value = element.getAttribute("Class");
		System.out.println("Color of the button: " + att_value);
		
		//slate button and print it's text.
		WebElement element2 = driver.findElement(By.xpath("//button[contains(text(), 'Sl')]"));
		String button_text = element2.getText();
		System.out.println("text of Slate button: " + button_text);
		
		//Close the browser
		driver.quit();
		
	}

}
